### PR TITLE
fix(all): F5 always re-auth and manage session lost

### PR DESCRIPTION
**Pull request by guillaumechervetaxa**
#317 and #301

### CODE REVIEW

**Comment by youf-olivier** [comment]
Even we are an autosquash, clean commits are nicer to follow you on the PR. (for next time ;))

File: packages/core/src/services/oidcServices.js [hunk 1]
```diff
@@ -1,20 +1,39 @@
...
-  if (isRequireAuthentication(oidcUser, isForce) && !userRequested) {
+  const url = location.pathname + (location.search || '');
+
+  if (isRequireSignin(oidcUser, isForce)) {
     oidcLog.info('authenticate user...');
-    const url = location.pathname + (location.search || '');
     userRequested = true;
     await userManager.signinRedirect({ data: { url } });
     userRequested = false;
+  } else if (oidcUser && oidcUser.expired) {
```
Comment (youf-olivier) [inline packages/core/src/services/oidcServices.js]: We never had to do that, don't you think that the problem is deeper ?

Comment (guillaume-chervet) [reply packages/core/src/services/oidcServices.js]: Je vais essayer de finir cette pr la semaine prochaine. J'avais presque fini.

**Comment by Shashank-11** [comment]
Any chance of merging this PR?

**Comment by youf-olivier** [comment]
Yes. We have 3 PR to merge. December was a very high intensive month, and for now Guillaume is on vacation. We close all PR start of January

File: packages/core/src/services/oidcServices.js (hunk 1 above)
Comment (guillaumechervetaxa) [reply packages/core/src/services/oidcServices.js]: Oui c'est ce qui uniformise le comportement des 2 librairies.

**Comment by guillaumechervetaxa** [comment]
Ready for review @youf-olivier,

I do not remember the property you used to shorten the accesstoken lifetime for testing.
Do you know which property it is?

**Comment by youf-olivier** [comment]
> Ready for review @youf-olivier,
> 
> I do not remember the property you used to shorten the accesstoken lifetime for testing.
> Do you know which property it is?

It is the client id in config file. With IS4 demo you can set the client Id to spa.short to get a shorlive token.

**Pull request merged**